### DISCUSSION TITLE
Updating pip packages

### DISCUSSION THREAD

**Discussion by tomas-silveira** [sebp/scikit-survival#296] [General]
Hello sebp, would it be possible to update the pip package to include more versions>0.14.0? I've moved to an dev environment where I'm unable to install packages from conda, but am able to install them using pip (specifically I would need v0.17.0 if possible). I've also tried to install it by cloning v0.17.0 and running `python setup.py install` but am also getting an error:

<img width="1374" alt="1" src="https://user-images.githubusercontent.com/41126502/188455298-d1483252-676b-41a3-a530-98ff42fbc431.PNG">

Thank you in advance.

**Comment by sebp** [comment]
I'm using https://github.com/pypa/cibuildwheel to automatically build wheels for each release. I might be able to run it locally on a Linux machine to build wheels for older versions.

**Comment by sebp** [comment]
I just uploaded wheels for 0.17.0 to pypi, please give it a try.

**Comment by sebp** [comment]
Regarding the error in your screenshot, you need to clone submodules too, see https://scikit-survival.readthedocs.io/en/latest/contributing.html#creating-a-fork

**Comment by tomas-silveira** [comment]
Much appreciated @sebp, thank you. However, since I'm locked on python v3.6.10 because of another package (pyspark 2.2.0), I'm unable to install sksurv with version 0.17.0 because it requires python>3.7...
I've also tried installing it by cloning (and cloning submodules too, as per your answer), then changing the requirements to be python>3.6 instead of 3.7, but I'm getting an error when running `python setup.py install`:

<img width="1067" alt="1" src="https://user-images.githubusercontent.com/41126502/188605788-7ba18520-f921-4d46-b307-a9e89a997efa.PNG">

**Reply by sebp** [reply]
If you try to install sksurv after manually requiring Python >= 3.6, it probably will pick up a wrong version of scikit-learn, which isn't supported with 0.17.0. Moreover, scikit-learn 1.0, which is supported, requires Python >= 3.7 too. I'm afraid, supporting 3.6 would be rather difficult.

**Reply by tomas-silveira** [reply]
No problem, I totally understand. I'll see if I can solve it another way. Thank you @sebp
